odoo/cli/command.py

```diff
@@ -17,54 +20,79 @@ class Command:
 
     def __init_subclass__(cls):
         cls.name = cls.name or cls.__name__.lower()
+        module = cls.__module__.rpartition('.')[2]
+        if not cls.is_valid_name(cls.name):
+            raise ValueError(
+                f"Command name '{cls.name!r}' "
+                f"must match '{COMMAND_NAME_RE.pattern}'")
+        if cls.name != module:
+            raise ValueError(
+                f"Command name '{cls.name!r}' "
+                f"must match Module name '{module!r}'")
         commands[cls.name] = cls
 
+    @classmethod
+    def is_valid_name(cls, name):
+        return re.match(COMMAND_NAME_RE, name)
+
 
 def load_internal_commands():
-    """Load `commands` from `odoo.cli`"""
+    """Load ``commands`` from ``odoo.cli``"""
     for path in odoo.cli.__path__:
         for module in Path(path).iterdir():
             if module.suffix != '.py':
                 continue
             __import__(f'odoo.cli.{module.stem}')
 
 
-def load_addons_commands():
-    """Load `commands` from `odoo.addons.*.cli`"""
-    logging.disable(logging.CRITICAL)
-    initialize_sys_path()
-    for module in get_modules():
-        if (Path(get_module_path(module)) / 'cli').is_dir():
-            with contextlib.suppress(ImportError):
-                __import__(f'odoo.addons.{module}')
-    logging.disable(logging.NOTSET)
-    return list(commands)
+def load_addons_commands(paths, search_term="*"):
+    """
+        Load ``commands`` from ``odoo.addons.<app>.cli.<command>``
+        to ``odoo.cli.<command>``. We need to change the qualified
+        name because it would load the app's root ``__init__.py``
+        and import too many things.
+
+        If None or '*' is used as a search_term, all commands
+        are found and loaded, for example, for `help` purposes.
+    """
```

Comment:
We don't usually add a new indentation inside the docstring, we align the text with the opening/closing triple quotes. The part about `__init__.py` is an implementation detail and shall be moved out of the docstring.